src/libostree/ostree-sysroot-deploy.c

```diff
@@ -3722,11 +3726,22 @@ _ostree_sysroot_finalize_staged_inner (OstreeSysroot *self, GCancellable *cancel
     }
 
   /* Check if finalization is locked. */
-  if (!glnx_fstatat_allow_noent (AT_FDCWD, _OSTREE_SYSROOT_RUNSTATE_STAGED_LOCKED, NULL, 0, error))
-    return FALSE;
-  if (errno == 0)
+  gboolean locked = false;
+  (void)g_variant_lookup (self->staged_deployment_data, _OSTREE_SYSROOT_STAGED_KEY_LOCKED, "b",
+                          &locked);
+  if (locked)
+    g_debug ("staged is locked via metadata");
+  else
+    {
+      if (!glnx_fstatat_allow_noent (AT_FDCWD, _OSTREE_SYSROOT_RUNSTATE_STAGED_LOCKED, NULL, 0,
```

Comment:
As part of stabilization, shouldn't we make this file path public? And document how to unlock it.

Also, should we add `ostree admin finalize-staged --unlock`?

Reply:
Right thanks, this was not fully baked.  I've updated this more, though needs some testing.

Reply:
> As part of stabilization, shouldn't we make this file path public? 

Here we're moving the locked state into the staged deployment data itself (which would have avoided the earlier problem) but it does mean we need an API and CLI to mutate that state.